packages/dds/tree/src/simple-tree/treeNodeApi.ts

```diff
@@ -171,13 +176,23 @@ export const treeNodeApi: TreeNodeApi = {
 				return unreachableCase(eventName);
 		}
 	},
-	status: (node: TreeNode) => {
-		return getFlexNode(node, true).treeStatus();
+	status(node: TreeNode): TreeStatus {
+		const flexNode = getFlexNode(node, true);
+		if (isMapTreeNode(flexNode)) {
+			return TreeStatus.New;
+		}
+
+		assert(flexNode instanceof LazyEntity, "Unexpected flex node implementation");
+		if (flexNode[isFreedSymbol]()) {
+			return TreeStatus.Deleted;
+		}
+
+		return treeStatusFromAnchorCache(flexNode.anchorNode);
```

Comment:
this function has a really weird name. why is what you pass an anchor cache?
that aside, I assume we can inline that implementation when we remove flex tree?

Reply:
Ha, yea. `treeStatusFromAnchorCache` and friends is the code the caches the detached field on the anchor slot (without the cache, it's an O(depth) walk every to find the field from the path every time). I attempted to move it up out of flex tree in this PR but it is still needed in flex tree (solely) to emit the user error for trying to mutate a tree that is not in the document (see the other comment I made on this PR).